eks/ng/role.go

```diff
@@ -205,8 +208,30 @@ Outputs:
 
 `
 
+type templateRole struct {
+	UserData string
+}
+
+const userDataASGPolicy = `          - Effect: Allow
+            Action:
+            - autoscaling:DescribeAutoScalingGroups
+            - autoscaling:DescribeAutoScalingInstances
+            - autoscaling:DescribeLaunchConfigurations
+            - autoscaling:DescribeTags
+            - autoscaling:SetDesiredCapacity
+            - autoscaling:TerminateInstanceInAutoScalingGroup
+            - ec2:DescribeLaunchTemplateVersions
+            Resource: "*"
+`
+
 func (ts *tester) createRole() error {
 	if !ts.cfg.EKSConfig.AddOnNodeGroups.RoleCreate {
+		// if role isn't created and autoscaler is enabled then return error
+		// because we cannot assume ASG policy is attached to node iam role
+		// this check is also enabled when calling createCA() in nodes.go which comes after createRole() func call
+		if ts.cfg.EKSConfig.IsEnabledAddOnClusterAutoscaler() {
```

Comment:
Remove `ts.cfg.EKSConfig.IsEnabledAddOnClusterAutoscaler` since we are adding autoscaler inside node group `struct`. You can iterate `NGs` and store a boolean variable to indicate whether to add ASG policy or not.

Reply:
Sure, that makes more sense ( as we are planning to move it inside the NG)